### PR TITLE
[feat] [style] Create and Style Filters Dropdown

### PR DESCRIPTION
**Pull request by ethan-tam33**
### Description
[//]: # "[Required] Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- fixed the filter buttons in `FilterBar` to  be individual
- created `MapViewScreen` component
- created and styled `TechnologyDropdown` component
- enabled `TechnologyDropdown` component to render on top of the "Technology" filter button upon clicking "Technology" filter button
- created `Filters` type to hold all possible changes from "Status", "Technology", "Project Size", and "Location" filter dropdowns
- changed react-icon images to SVG images directly from Figma design

Some relevant stuff I created:
`handleButtonClick` - a function that takes in a filter and allows for 1. toggling the dropdown bar and 2. any other action we want to implement
`filterChangeHandlers` - a variable that holds all the handler methods for each filter
`Filter` - a component that will hold all the dropdowns of each filter

### Screenshots
[//]: # "Add any screenshots that will help your PR reviewer understand what’s going on in the PR!"
[//]: # "[Required] For frontend tasks: add screenshots of the layout and expected behavior (GIFs if you’re feeling extra fancy)!"
[//]: # "Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

Screenshot 2024-11-11 at 6 30 16 AM

## How to review
[//]: # "/[Required] Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

## Next steps
[//]: # "[Optional] - What has not yet been added to the PR, doesn’t yet have any functionality — essentially, what needs to be fixed or added."
[//]: # "[Optional] any temporary fixes in this PR that should be cleaned up later."

- create and style dropdowns for status, project size, and location filters

### Online sources
[//]: # "[Optional] insert links to any useful tutorials or documentation that you used for this PR!"

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @itsliterallymonique

### CODE REVIEW

File: components/FilterBar/index.tsx [hunk 1]
```diff
@@ -16,20 +17,58 @@ export const FilterBar = ({
...
-          <FilterButtonStyles
-            key={filter.label}
-            onClick={() => onFilterChange(filter)}
-          >
-            {filter.icon}
-            {filter.label}
-            <RiArrowDropDownLine />
-          </FilterButtonStyles>
-        </div>
+        <FilterBackgroundStyles
```
Comment (itsliterallymonique) [inline components/FilterBar/index.tsx:47]: can you turn this into a `Filter` component. Then you pass in the following props: label, isActive, icon, dropdown component (or what props you need to decide which filter it is). Right now the filter bar component looks messy. This will help further abstract and organize the different filters so that it's easier for people to make their own filter dropdowns.

**Review by itsliterallymonique — CHANGES_REQUESTED** [review]
Made one comment on making a separate filter component. Will comment on the styling once you push your final changes

File: components/Filter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  selectedFilters: Filters;
+  onTechnologyChange: (options: string[]) => void;
+  handleButtonClick: (filter: FilterType) => void;
+}
+
+export default function Filter({
+  filter,
+  isActive,
+  selectedFilters,
+  onTechnologyChange,
```
Comment (itsliterallymonique) [inline components/Filter/index.tsx:23]: change the name of this to something that is general for all filters.

Comment (itsliterallymonique) [reply components/Filter/index.tsx:23]: e.g. onFilterChange?

File: components/Filter/index.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            handleButtonClick={handleButtonClick}
+            icon={filter.icon}
+            label={filter.label}
+            currFilter={filter}
+          />
+        ) : // Add other filter dropdown components here
+        null
+      ) : (
+        <FilterButtonStyles onClick={() => handleButtonClick(filter)}>
+          <IconStyle>{filter.icon}</IconStyle>
```
Comment (itsliterallymonique) [inline components/Filter/index.tsx:42]: the icon is wrong. It should be the energy icon
Screenshot 2024-11-12 at 5 23 46 PM

File: components/FilterBar/styles.ts [hunk 4]
```diff
@@ -1,9 +1,17 @@
...
 import { FilterHeadingUnused } from '@/styles/texts';
 
-export const FilterBarStyles = styled.div`
+export const FilterContainerStyles = styled.div`
   display: flex;
-  position: absolute;
+  flex-direction: row;
+  justify-content: flex-end;
+  width: 100%;
+  margin-right: 15%;
```
Comment (itsliterallymonique) [inline components/FilterBar/styles.ts:9]: make this ~3rem

File: components/Filter/styles.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import styled from 'styled-components';
+import { FilterHeadingUnused } from '@/styles/texts';
+
+export const FilterBackgroundStyles = styled.div<{ isActive: boolean }>`
+  margin-right: 2%;
```
Comment (itsliterallymonique) [inline components/Filter/styles.ts:5]: 0.375rem instead. The gap is bigger than josh's figma hi-fis

File: components/Filter/styles.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  background: linear-gradient(
+    180deg,
+    rgba(250, 250, 250, 0.32) 0%,
+    rgba(238, 238, 238, 0.65) 100%
+  );
+  backdrop-filter: blur(7.5px);
+  padding: 0.35rem 0.35rem;
+  z-index: 5;
+  border: 0.05rem solid #fff;
+  margin-top: 1.5%;
```
Comment (itsliterallymonique) [inline components/Filter/styles.ts:17]: 0.75 rem

**Review by itsliterallymonique — CHANGES_REQUESTED** [review]
Make the quick styling changes I commented on. Then you're good to squash + merge!

**Comment by ethan-tam33** [comment]
I realized some parts of the Filter component wasn't generalizable, specifically the handlers. I created`FilterChangeHandlers` to hold all the handler method of each filter.

File: components/Filter/index.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    <FilterBackgroundStyles isActive={isActive}>
+      {isActive ? (
+        filter.id === 'technology' ? (
+          <TechnologyDropdown
+            selectedTechnologies={selectedFilters.technology}
+            setSelectedTechnologies={filterChangeHandlers.technology}
+            handleButtonClick={handleButtonClick}
+            icon={filter.icon}
+            label={filter.label}
+            currFilter={filter}
```
Comment (itsliterallymonique) [inline components/Filter/index.tsx:36]: what does currFilter do?

File: components/Filter/styles.ts (hunk 5 above)
Comment (itsliterallymonique) [inline components/Filter/styles.ts:5]: i think you have to make the changes here and below where I commented. The styling still does not look like josh's figma hi-fis
